### PR TITLE
[docs] add TenTap Rich Text Editor to docs

### PR DESCRIPTION
**Pull request by 17Amir17**
# Why

[TenTap](https://github.com/10play/10Tap-Editor) is an open source WebView based rich text editor I have been working on, and we just added expo support!

# How

I added it to the section where WebView based solutions are added, but TenTap also exposes a way to create custom "editor build" so I added a description for that too.

### CODE REVIEW

**Comment by GuySerfaty** [comment]
@Simek / @amandeepmittal - any chance you take a look on this one?

**Comment by kadikraman** [comment]
Hi @GuySerfaty thank you for the PR! I recall seeing your demo of it on Discord a while ago and it looked great!

I've just tried it out locally and I had a question regarding the documentation:

The toolbar doesn't actually show up using the code in the [readme](https://github.com/10play/10tap-editor/tree/main?tab=readme-ov-file#usage). When I looked at the [basic example](https://github.com/10play/10tap-editor/blob/main/example/src/Examples/Basic.tsx#L21), I found that you're actually wrapping it in a `KeyboardAvoidingView` and positioning it absolutely. Could that code be included in the example? I believe most people would expect the copy/paste-able example to just work out of the box.

Also as a side note: the link in the readme to the basic example is broken btw, it links to a [page that doesn't exist](https://github.com/10play/10tap-editor/blob/examples/basic.md). I believe it's supposed to link [here](https://github.com/10play/10tap-editor/blob/main/example/src/Examples/Basic.tsx).

File: docs/pages/guides/editing-richtext.mdx [hunk 1]
```diff
@@ -47,6 +47,7 @@ There are a couple of existing React Native libraries to allow rich-text editing
 
 - [`react-native-rich-editor`](https://github.com/wxik/react-native-rich-editor)
 - [`react-native-cn-quill`](https://github.com/imnapo/react-native-cn-quill)
+- [`@10play/tentap-editor`](https://github.com/10play/10Tap-Editor) - TenTap also gives you a way to customize the editor with some additional steps.
```
Comment (amandeepmittal) [inline docs/pages/guides/editing-richtext.mdx:50]:
```suggestion
- [`@10play/tentap-editor`](https://github.com/10play/10Tap-Editor)
```

We don't add descriptions to each library in this doc so let's follow the same pattern.

**Comment by 17Amir17** [comment]
> Hi @GuySerfaty thank you for the PR! I recall seeing your demo of it on Discord a while ago and it looked great!
> 
> I've just tried it out locally and I had a question regarding the documentation:
> 
> The toolbar doesn't actually show up using the code in the [readme](https://github.com/10play/10tap-editor/tree/main?tab=readme-ov-file#usage). When I looked at the [basic example](https://github.com/10play/10tap-editor/blob/main/example/src/Examples/Basic.tsx#L21), I found that you're actually wrapping it in a `KeyboardAvoidingView` and positioning it absolutely. Could that code be included in the example? I believe most people would expect the copy/paste-able example to just work out of the box.
> 
> Also as a side note: the link in the readme to the basic example is broken btw, it links to a [page that doesn't exist](https://github.com/10play/10tap-editor/blob/examples/basic.md). I believe it's supposed to link [here](https://github.com/10play/10tap-editor/blob/main/example/src/Examples/Basic.tsx).

Sure this makes sense! I'll update

**Review by amandeepmittal — APPROVED** [review]

**Review by kadikraman — APPROVED** [review]

**Pull request merged**